### PR TITLE
AEA-856 multiple doc updates, incl docker image ex

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR addresses a few minor doc requests.

## Fixes

- https://github.com/fetchai/agents-aea/issues/1121
- https://github.com/fetchai/agents-aea/issues/803
- https://github.com/fetchai/agents-aea/issues/1382

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: docs/api/decision_maker/messages/base.md [hunk 1]
```diff
@@ -1,117 +0,0 @@
-<a name=".aea.decision_maker.messages.base"></a>
-# aea.decision`_`maker.messages.base
-
-This module contains the base message and serialization definition.
```
Comment (DavidMinarsch) [inline docs/api/decision_maker/messages/base.md:4]: Old code; removed

File: docs/api/decision_maker/messages/state_update.md [hunk 2]
```diff
@@ -1,98 +0,0 @@
-<a name=".aea.decision_maker.messages.state_update"></a>
-# aea.decision`_`maker.messages.state`_`update
-
```
Comment (DavidMinarsch) [inline docs/api/decision_maker/messages/state_update.md:3]: Old code; removed

File: docs/api/decision_maker/messages/transaction.md [hunk 3]
```diff
@@ -1,286 +0,0 @@
-<a name=".aea.decision_maker.messages.transaction"></a>
-# aea.decision`_`maker.messages.transaction
-
```
Comment (DavidMinarsch) [inline docs/api/decision_maker/messages/transaction.md:3]: Old code; removed

File: docs/deployment.md [hunk 4]
```diff
@@ -1,9 +1,39 @@
...
+## Building a Docker Image
+
+First, we fetch a directory containing a Dockerfile and some dependencies:
+``` bash
+svn export https://github.com/fetchai/agents-aea/branches/master/deploy-image
+cd deploy-image
+rm -rf scripts
+svn export https://github.com/fetchai/docker-images/branches/master/scripts
+cd ..
+```
```
Comment (DavidMinarsch) [inline docs/deployment.md:17]: This can eventually be streamlined; but as a first step this works

File: docs/glossary.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+This glossary defines a number of common terms used throughout the documentation. For the definition of framework components consult the API docs.
+
+* AEA: an Autonomous Economic Agent (AEA) is a "an intelligent agent acting on an owner's behalf, with limited or no interference, and whose goal is to generate economic value to its owner". AEAs are a special type of agent.
+
+* (Software) Agent: a software agent is a computer program that acts on behalf of an entity (e.g. individual, organisation, business).
```
Comment (DavidMinarsch) [inline docs/glossary.md:5]: We can add to the glossary as we go along.

**Review by marcofavorito — APPROVED** [review]
LGTM

File: deploy-image/entrypoint.sh [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
 set -e
 
+aea --version
+
 if [ -z ${AGENT_REPO_URL+x} ] ; then
         rm myagent -rf
-        aea create myagent
-        cd myagent
-        aea add skill fetchai/echo:0.6.0
+        aea fetch fetchai/my_first_aea:0.10.0
```
Comment (solarw) [inline deploy-image/entrypoint.sh:8]: probably `add fetch.. && cd .. ` ?

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**